File: src/main/java/com/fobgochod/git/commit/view/settings/TypeEditor.java
package com.fobgochod.git.commit.view.settings;

import com.intellij.openapi.ui.DialogWrapper;
import com.intellij.util.ui.FormBuilder;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;
import java.awt.*;

public class TypeEditor extends DialogWrapper {

    private final JPanel root;
    private final JTextField titleField;
    private final JTextField descriptionField;

    public TypeEditor(String dialogTitle, String title, String description) {
        super(true);
        setTitle(dialogTitle);
        
        root = new JPanel();
        titleField = new JTextField(title);
        descriptionField = new JTextField(description);

        JPanel panel = FormBuilder.createFormBuilder()
                .setHorizontalGap(5)
                .addLabeledComponent(new JLabel("title"), titleField)
                .addLabeledComponent(new JLabel("description"), descriptionField)
                .getPanel();
        panel.setPreferredSize(new Dimension(460, 70));
        root.add(panel);

        init();
    }

    public String getTitle() {
        return titleField.getText().trim();
    }

    public String getDescription() {
        return descriptionField.getText().trim();
    }

    @Nullable
    @Override
    protected JComponent createCenterPanel() {
        return root;
    }

    public interface Validator {
        boolean isOK(String name, String value);
    }
}

File: src/main/kotlin/com/fobgochod/git/commit/view/TypeEditor.kt
package com.fobgochod.git.commit.view

import com.fobgochod.git.commit.util.GitBundle
import com.intellij.openapi.ui.DialogWrapper
import com.intellij.ui.components.JBTextField
import com.intellij.util.ui.FormBuilder
import java.awt.BorderLayout
import java.awt.Dimension
import javax.swing.JComponent
import javax.swing.JLabel
import javax.swing.JPanel

class TypeEditor(title: String, name: String, description: String) : DialogWrapper(true) {

    private val root: JPanel
    private val nameLabel = JLabel(GitBundle.message("settings.type.table.column.1"))
    private val nameField: JBTextField
    private val descriptionLabel = JLabel(GitBundle.message("settings.type.table.column.2"))
    private val descriptionField: JBTextField

    init {
        setTitle(title)
        root = JPanel(BorderLayout())
        nameField = JBTextField(name)
        descriptionField = JBTextField(description)
        val formBuilder = FormBuilder.createFormBuilder()
            .addLabeledComponent(nameLabel, nameField)
            .addLabeledComponent(descriptionLabel, descriptionField)

        root.add(formBuilder.panel, BorderLayout.CENTER)
        root.preferredSize = Dimension(500, 0)
        init()
    }

    fun name(): String = nameField.text.trim()
    fun description(): String = descriptionField.text.trim { it <= ' ' }

    override fun createCenterPanel(): JComponent {
        return root
    }
}
